src/Model/Documentable/DocumentableProductTrait.php

```diff
@@ -229,28 +202,30 @@ protected function addOptionsInDocument(ResultInterface $document, string $local
         return $document;
     }
 
-    /**
-     * @param $channel
-     *
-     * @return null
-     */
-    private function getCheapestVariantForChannel($channel)
+    private function getCheapestVariantChannelPricingForChannel(ChannelInterface $channel): ?ChannelPricingInterface
     {
-        $cheapestVariant = null;
-        $cheapestPrice = null;
+        $cheapestVariantChannelPricing = null;
+        $cheapestProductPrice = null;
         $variants = $this->getEnabledVariants();
         foreach ($variants as $variant) {
             $channelPrice = $variant->getChannelPricingForChannel($channel);
-            if (null === $cheapestPrice || $channelPrice->getPrice() < $cheapestPrice) {
-                $cheapestPrice = $channelPrice->getPrice();
-                $cheapestVariant = $variant;
+            if ($channelPrice === null) {
+                continue;
+            }
+            $variantPrice = $channelPrice->getPrice();
+            if ($variantPrice < $channelPrice->getMinimumPrice()) {
+                $variantPrice = $channelPrice->getMinimumPrice();
+            }
```

Comment:
```suggestion
            $variantPrice = min($channelPrice->getPrice(), $channelPrice->getMinimumPrice());
```
Again, it looks to me that the original check was backward.

Reply:
We don't have to check for the min price between the price and minimumPrice, we have to take the minimum price only if the price is less than this.